### PR TITLE
[NETBEANS-2443] [NETBEANS-5732] Support profiler or other StartupExtenders that use spaces in args

### PR DESCRIPTION
**Pull request by sdedic**
An alternative to #3077 - an attempt to fix flaky`StartupExtender` specification. The main issue is that while `StartupExtender` API provides `List` of arguments, the API did not make it explicit that no quoting should be used - that the returned list items contain the verbatim conents that should be passed to the VM.  In general quoting should be done **always by the launcher**: it can put individual arguments right into the `ProcessBuilder` or somehow quote/concatenate into a commandline string.

I've added a flag that allows a `StartupExtender` to declare it does/does not quoting. `getRawArguments` (aded API method) will then contain literal parameter value regardless of the extender policy. For compatibity, the `getArguments` method is unchanged.

I have adapted maven and gradle. Maven has more tweaks as I found multiple issues with space or quote handling in Maven / Windows.

This should fix [NETBEANS-2443](https://issues.apache.org/jira/browse/NETBEANS-2443) and [NETBEANS-5732](https://issues.apache.org/jira/browse/NETBEANS-5732)

Unit tests added.

### CODE REVIEW

**Comment by moacirrf** [comment]
Thanks man, great job, I was taking days trying to solve it, I will put your name on the task 2443 jira, ok?
Thanks.

**Comment by sdedic** [comment]
@moacirrf If you have some spare time, try to build from the branch & test on Windows. I did some basics, a colleague who runs Windows computer tested as well, but still....

**Comment by moacirrf** [comment]
Tested on windows:
Maven: Working
Gradle: Working
Netbeans Project(Ant): Not working for me when netbeans is on a path with spaces.

**Correct directory**
Z:/java/Netbeans/Program Files/...

**Error:**
Error occurred during initialization of VM
Could not find agent library Z:/java/Netbeans/Program in absolute path, with error: Can't find dependent libraries

**Comment by sdedic** [comment]
@moacirrf Thanks for testing! I've improved compatibility (added tests to startup extender API module); I've also reviewed StartupExtender impls in the current repository:

Fixed: `webcommon/javascript.nodejs/src/org/netbeans/modules/javascript/nodejs/exec/NodeExecutable.java`
Mixed quoted arguments (getArguments) and unquoted, then uses list of such mixed args for ProcessBuilder

Concatenate arguments into a single property - can continue to use getArguments, no change needed:
* `contrib/j2ee.jboss4/src/org/netbeans/modules/j2ee/jboss4/ide/JBStartRunnable.java`
* `java/java.source.ant/src/org/netbeans/modules/java/source/ant/ProjectRunnerImpl.java`
* `apisupport/maven.apisupport/src/org/netbeans/modules/maven/apisupport/NetBeansStartupArgs.java`
* `apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/ui/ModuleActions.java`
* `javafx/javafx2.project/src/org/netbeans/modules/javafx2/project/JFXActionProvider.java`
* `enterprise/j2ee.earproject/src/org/netbeans/modules/j2ee/earproject/EarActionProvider.java`
* `enterprise/tomcat5/src/org/netbeans/modules/tomcat5/optional/StartTomcat.java`

Just adds a fixed parameter without spaces - no change needed.
`java/debugger.jpda.trufflenode/src/org/netbeans/modules/debugger/jpda/truffle/node/GraalVmStartupExtender.java`

Potential bug:
* `enterprise/glassfish.common/src/org/netbeans/modules/glassfish/common/StartTask.java`
* `enterprise/payara.common/src/org/netbeans/modules/payara/common/StartTask.java`
filed https://issues.apache.org/jira/browse/NETBEANS-5908

**Comment by sdedic** [comment]
@moacirrf I believe I've fixed the case for Ant projects (and for some others). It would be great if you could re-test to be sure.

**Review by JaroslavTulach — APPROVED** [review]

File: ide/extexecution/build.xml [hunk 1]
```diff
@@ -22,4 +22,14 @@
...
     <import file="../../nbbuild/templates/projectized.xml"/>
+    
+    <!-- 
+        Hacks the build process: we supply a default generated-layer, the layer reg processor reads the existing 
+        file first, and appends to it. So annotation-based registrations will merge into the static content.
+        Must be done before do-test-compile.
+    -->
+    <target name="test-unit-dep-build" depends="test-init,projectized-common.test-unit-dep-build">
+       <mkdir dir="${build.test.unit.classes.dir}/META-INF"/>
+       <copy file="${test.unit.src.dir}/META-INF/layer-static.xml" tofile="${build.test.unit.classes.dir}/META-INF/generated-layer.xml"/>
```
[the remark this answers is not in the document]
Comment (sdedic) [reply ide/extexecution/build.xml]: Fixed in 48182cf

**Comment by moacirrf** [comment]
Tested Ant, its working.
thanks

**Comment by sdedic** [comment]
@moacirrf awesome - thanks for the verification and support !

**Comment by sdedic** [comment]
added @tmysik to review the change in `NodeExecutable`.

**Review by tmysik — COMMENTED** [review]
`NodeExecutable` class should be fine, I think. Thanks!

**Pull request merged**